figures/serializers.py

```diff
@@ -158,6 +158,12 @@ def get_courses(self, obj):
         ).filter(
             user=obj.user
         ).using(read_replica_or_default())
+
+        if obj.course_id:
+            course_enrollments = course_enrollments.filter(
+                course_id=as_course_key(obj.course_id)
+            )
```

Comment:
Add read replica support `.using(read_replica_or_default())`

Reply:
`.using(read_replica_or_default())` is already added in the query. It was a missing condition added.